airbyte-integrations/connectors/source-cockroachdb/src/main/java/io/airbyte/integrations/source/cockroachdb/CockroachDbSource.java

```diff
@@ -110,12 +110,29 @@ protected boolean isNotInternalSchema(JsonNode jsonNode, Set<String> internalSch
     return false;
   }
 
+  @Override
+  public JdbcDatabase createDatabase(final JsonNode config) throws SQLException {
+    final JsonNode jdbcConfig = toDatabaseConfig(config);
+
+    final JdbcDatabase database = Databases.createJdbcDatabase(
+            jdbcConfig.get("username").asText(),
+            jdbcConfig.has("password") ? jdbcConfig.get("password").asText() : null,
+            jdbcConfig.get("jdbc_url").asText(),
+            driverClass,
+            jdbcConfig.has("connection_properties") ? jdbcConfig.get("connection_properties").asText() : null,
+            sourceOperations);
+
+    quoteString = (quoteString == null ? database.getMetaData().getIdentifierQuoteString() : quoteString);
+
+    return new CockroachJdbcDatabase(database, sourceOperations);
+  }
+  
   private CheckedFunction<Connection, PreparedStatement, SQLException> getPrivileges(JdbcDatabase database) {
     return connection -> {
       final PreparedStatement ps = connection.prepareStatement(
           "SELECT DISTINCT table_catalog, table_schema, table_name, privilege_type\n"
               + "FROM   information_schema.table_privileges\n"
-              + "WHERE  grantee = ? AND privilege_type in ('SELECT', 'ALL')");
+              + "WHERE  (grantee  = ? AND privilege_type in ('SELECT', 'ALL')) OR (table_schema = 'public')");
```

Comment:
This change is unrelated to the bug, but without this, tables in the `public` schema are not visible in the Airbyte UI (which feels odd, since tables in the `public` schema in CockroachDB are visible to all users regardless of explicit grants). Feel free to undo this if this isn't a change you'd like to incorporate in the connector